lib/pi_piper/bcm2835.rb

```diff
@@ -1,5 +1,15 @@
-require "pi_piper/bcm2835/version"
+require 'pi_piper/bcm2835/version'
+require 'pi_piper/bcm2835/pin'
+require 'pi_piper/bcm2835/spi'
+require 'pi_piper/bcm2835/i2c'
+require 'pi_piper/bcm2835/driver'
 
 module PiPiper
-  autoload PiPiper::Bcm2835, "pi_piper/bcm2835/bcm2835"
+  module Bcm2835
+    class << self
+      def driver
+        @driver ||= PiPiper::Bcm2835::Driver.new
+      end
```

Comment:
I think it is a missunderstanding.
PiPiper.driver in my implementation is the only instance of any driver kind loaded at any moment.
I even though a moment to use the singleto module : http://ruby-doc.org/stdlib-1.9.3/libdoc/singleton/rdoc/Singleton.html

If we want to be able to have several drivers for several Object, we have to rethink the implementation. I don't think it is very useful, but we can discuss it.